stablehlo/tests/bytecode.mlir

```diff
@@ -0,0 +1,780 @@
+  %0:2 = "stablehlo.rng_bit_generator"(%arg) {rng_algorithm = #stablehlo.rng_algorithm<PHILOX>} : (tensor<3xui64>) -> (tensor<3xui64>, tensor<2x2xui32>)
+  %1 = "stablehlo.tuple"(%0#0, %0#1) : (tensor<3xui64>, tensor<2x2xui32>) -> tuple<tensor<3xui64>, tensor<2x2xui32>>
+  func.return %1 : tuple<tensor<3xui64>, tensor<2x2xui32>>
+}
+
+func.func @test_cbrt(%arg: tensor<3x4xf32>) -> tensor<3x4xf32> {
+  %0 = "stablehlo.cbrt"(%arg) : (tensor<3x4xf32>) -> tensor<3x4xf32>
+  func.return %0 : tensor<3x4xf32>
+}
+
+func.func @test_reduce_precision(%arg: tensor<3x4xf32>) -> tensor<3x4xf32> {
+  %0 = "stablehlo.reduce_precision"(%arg) {exponent_bits = 8 : i32, mantissa_bits = 10 : i32} : (tensor<3x4xf32>) -> tensor<3x4xf32>
+  func.return %0 : tensor<3x4xf32>
+}
+
+func.func @test_optimization_barrier(%arg0: tensor<4x4xf32>, %arg1: tensor<3x4xf32>) -> (tensor<4x4xf32>, tensor<3x4xf32>) {
+  %0, %1 = "stablehlo.optimization_barrier"(%arg0, %arg1) : (tensor<4x4xf32>, tensor<3x4xf32>) -> (tensor<4x4xf32>, tensor<3x4xf32>)
+  func.return %0, %1 : tensor<4x4xf32>, tensor<3x4xf32>
+}
+
+func.func @test_triangular_solve2(%arg0: tensor<4x4xf32>, %arg1: tensor<3x4xf32>) -> tensor<3x4xf32> {
+  %0 = "stablehlo.triangular_solve"(%arg0, %arg1) {left_side = false, lower = true, transpose_a = #stablehlo<transpose NO_TRANSPOSE>, unit_diagonal = false} : (tensor<4x4xf32>, tensor<3x4xf32>) -> tensor<3x4xf32>
+  func.return %0: tensor<3x4xf32>
+}
+
+func.func @test_reduce_window2(%arg0: tensor<4x2xf32>, %arg1: tensor<4x2xi32>, %init0: tensor<f32>, %init1: tensor<i32>) -> (tensor<2x2xf32>, tensor<2x2xi32>) {
+  %0:2 = "stablehlo.reduce_window"(%arg0, %arg1, %init0, %init1) ({
+         ^bb0(%a0: tensor<f32>, %a1: tensor<i32>, %b0: tensor<f32>, %b1: tensor<i32>):
+              %2 = stablehlo.add %a0, %b0 : tensor<f32>
+              %3 = stablehlo.add %a1, %b1 : tensor<i32>
+              "stablehlo.return"(%2, %3) : (tensor<f32>, tensor<i32>) -> ()
+            })
+         { padding = dense<[[2, 2], [0, 0]]> : tensor<2x2xi64>,
+           window_dimensions = dense<[5, 1]> : tensor<2xi64>,
+           window_strides = dense<[3, 1]> : tensor<2xi64> } : (tensor<4x2xf32>, tensor<4x2xi32>, tensor<f32>, tensor<i32>) -> (tensor<2x2xf32>, tensor<2x2xi32>)
+  func.return %0#0, %0#1 : tensor<2x2xf32>, tensor<2x2xi32>
+}
+
+func.func @test_round_nearest_even(%arg0: tensor<2xf32>) -> tensor<2xf32> {
+    %0 = "stablehlo.round_nearest_even"(%arg0) {} : (tensor<2xf32>) -> tensor<2xf32>
+    func.return %0 : tensor<2xf32>
+}
+
+// Test result alias
+func.func @test_alias_attribute (%arg0: tuple<tensor<f32>>
+    {stablehlo.result_alias = #stablehlo.result_alias<
+      tuple_indices = [0],
+      result_index = [0, 0],
+      must_alias>}
+    ) -> (tuple<tensor<f32>>) {
+  %0 = stablehlo.get_tuple_element %arg0[0] : (tuple<tensor<f32>>) -> tensor<f32>
+  %1 = stablehlo.tuple %0 : tuple<tensor<f32>>
+  func.return %1 : tuple<tensor<f32>>
+}
```

Comment:
That's amazingly thorough! My only two requests would be to: 1) sort alphabetically to simplify human analysis, 2) have function names match the tested entity, e.g. `func.func @cross_replica_sum` instead of `func.func @ test_crs` (although we cannot say `func.func @cross-replica-sum` without double quotes which looks weird).

Reply:
Agree - I actually had a slightly different plan for this. We have `stablehlo/tests/ops_chlo_roundtrip.mlir` which is just an error-free exhaustive test of chlo ops. I think `bytecode.mlir` would be better suited as `ops_stablehlo_roundtrip.mlir` and bytecode testing can be built on top of that. Similarly chlo bytecode testing is on top of `ops_chlo_roundtrip.mlir`. I'll do this migration in a followup change if that's ok?

Reply:
^ Reason that work is non-trivial is that I want to convert it to a FileCheck test in addition to a bytecode test, which involves a bit of modification.

Reply:
Will alphabetize the test point in the meantime, that part is simple

Reply:
Sounds like a plan!